Dump post processed XML file

rete opened the pull request:
BEGINRELEASENOTES
- Added write() method to XMLParser (resp. Parser) to write the loaded xml tree (resp. same file) in a new file.
- Many methods adapted in XMLParser to make parameter/constant replacements persistent in the XML tree
- New global parameter "OutputSteeringFile" introduced to write down a new steering file after processing
- Added command line option -n for a Marlin dry-run
- Example : 
```shell
# take the output file name from steering file and run Marlin as usual
Marlin steeringFile.xml 
# same but exit after writing the output steering file
Marlin -n steeringFile.xml 
# change the output file name
Marlin steeringFile.xml --global.OutputSteeringFile=MyParsedSteeringFile.xml 
```

ENDRELEASENOTES

gaede commented:
I believe what we discussed is:
 - introduce an optional global paramter (e.g. global.ParsedXMLFile )
- if this parameter is not empty write put the parsed XML file
- introduce a Marlin option to stop the processing after this parsing step
   - here we discussed both options '-p' for parse-only or '-n' for dry-run (either one would be fine with me)

rete commented:
I added a dry-run option -n
Summary : 
Printout the marlin XML and run Marlin : 
```shell
Marlin -p postSteering.xml steering.xml
```
Printout the marlin XML and exit : 
```shell
Marlin -n postSteering.xml steering.xml
```

@gaede : Is this OK for you in this way ? I could change it to put this functionnality in a xml parameter in the global section.

rete commented:
The output steering file name is no longer taken from command line args but from the input steering file itself by the key "OutputSteeringFile". This has the following consequences : 
- The write(fname) function in XMLParser was moved in the top-level class IParser and thus the method is also implemented in the Parser class (.steer files) as a raw copy paste of the input steering file
- The output steering file name can be specified by command line parameter using the usual parameter replacement mechanism

I added this key (OutputSteeringFile) to the steering file of the ctest that is testing the include mechanism. It works as expected.

rete commented:
I'm done with release notes

rete commented:
Do not merge yet. There are missing features in this PR that I discussed with Frank.

rete commented:
I added : 
- protection against nested include, by throwing exception if one is detected
- removed constant evaluation replacement in the XML tree for constants and processor parameters

For this latest point, it means that command line parameter are replaced in the constant element in the XML tree but not constant evaluation. As a consequence, the output parsed file will contain runtime argument from the command line while running in dry-run mode but will re-evaluate again the constants  and processor parameters in the next parsing. 

I don't know if I'm clear ...
